### DISCUSSION TITLE
How can I ignore a build step based on git information (e.g. branch name) without Vercel cloning the repo?

### DISCUSSION THREAD

**Discussion by michaelschufi** [vercel/community#7373] [Help] [answered]
### Summary

Hi :)

To reduce the waiting time for the builds in our monorepo, we're using an ignore build step script to not build a project if it's not needed. However, Vercel still spends about 15 seconds for each project in the monorepo cloning and running the ignore build step script, because the codebase grew in size. 

We would like to reduce those wait times by not cloning the repo on every check. Because 95% of the time, the cloning would not be necessary since we can check the commit metadata (e.g. branch name) to a environment variable to see if we actually need to build a project. So no code is required (besides the script atm).

### Steps to Reproduce

Our current script:

```ts
/* eslint-disable no-console */
const { VERCEL_GIT_COMMIT_REF, BRANCH_PREFIX } = process.env;

console.log(`BRANCH_PREFIX: ${BRANCH_PREFIX}`);
console.log(`VERCEL_GIT_COMMIT_REF: ${VERCEL_GIT_COMMIT_REF}`);

if (
  VERCEL_GIT_COMMIT_REF == 'dev' ||
  VERCEL_GIT_COMMIT_REF == 'main' ||
  VERCEL_GIT_COMMIT_REF.startsWith(`${BRANCH_PREFIX}/`)
) {
  // Proceed with the build
  console.log('✅ - Build can proceed');
  process.exit(1);
}

// Don't build
console.log('🛑 - Build cancelled');
process.exit(0);
```

**Comment by pawlean** [comment]
Hi @michaelschufi!

Have you checked out [Turborepo](https://vercel.com/docs/monorepos/turborepo)? I wonder if this is something you can try out in your project.

**Reply by pawlean** [reply]
Alternatively, have you got a `vercel.json` file that specifies your script above? e.g. something like:

```
{
  "build": {
    "ignore": "node ignore-build.js"
  }
} 
```

See [related documentation](https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel).

**Reply by michaelschufi** [reply]
Hi @pawlean 
Both solutions still clone the repo, correct? I want to reduce the time spent cloning, since that takes ~10-15s. So, depending on which project comes first, I have to wait `numberOfProjectsBeforeTheCorrectOne * 15s` for the build to start. Because every project will clone the repo first, and only afterwards execute the script. 

Edit: I'm already using Turborepo. But the script detecting if the build needs to run or not, is not the problem per se.

**Comment by michaelschufi** [comment, marked as answer]
This is probably solved by a new [Vercel feature](https://vercel.com/changelog/automatically-skip-unnecessary-deployments-in-monorepos). 

There's a discussion about it in the [Vercel Community board](https://vercel.community/t/how-can-i-ignore-a-build-step-based-on-git-information-e-g-branch-name-without-vercel-cloning-the-repo).

[a comment by an automated account is not shown]
